resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/CircuitBreakerStateMachine.java

```diff
@@ -942,6 +942,13 @@ public CircuitBreakerMetrics getMetrics() {
             this.permittedNumberOfCalls = new AtomicInteger(permittedNumberOfCallsInHalfOpenState);
             this.isHalfOpen = new AtomicBoolean(true);
             this.attempts = attempts;
+
+            final long waitDurationInHalfOpenState = circuitBreakerConfig.getWaitDurationInHalfOpenState().toMillis();
+            if (waitDurationInHalfOpenState >= 1000) {
+                ScheduledExecutorService scheduledExecutorService = schedulerFactory.getScheduler();
+                scheduledExecutorService
```

Comment:
I think this `ScheduledFuture` should be canceled if Circuit transit to other states. 
https://github.com/resilience4j/resilience4j/issues/991.

Reply:
Yes

Reply:
Yes, we introduces a preTransitionHook -> https://github.com/resilience4j/resilience4j/commit/378dd2b65889b83cda2a2b3d4203e86604b5044e